src/mongo/db/storage/csv_file.cpp

```diff
@@ -365,7 +364,25 @@ std::string_view CsvFileInput::getLine() {
     dassert(_offset <= _fileSize);
 
     size_t start = _offset;
-    while (_offset < _fileSize && _data[_offset] != '\n') {
+    bool quoteOpen = _data[_offset] == '\"';
+    if (quoteOpen) {
+        _offset++;
+    }
+
+    while (_offset < _fileSize && (_data[_offset] != '\n' || quoteOpen)) {
+        if (_data[_offset] == '\"') {
+            if (!quoteOpen && _data[_offset - 1] == ',') {  // if beginning of the field, open quote
+                quoteOpen = true;
+            }  // if quote is open and single quote, close the quote
+            else if (quoteOpen && _data[_offset + 1] != '\"') {
+                quoteOpen = false;
+            }  // if quote is open and double quote, extra increment _offset
+            else if (quoteOpen && _data[_offset + 1] == '\"') {
+                _offset++;
+            }
+            // Quote that appears in the unenclosed middle of field(aaa"""bb""cc"), or  appears
```

Comment:
[nit] extra space before appears.
```suggestion
            // Quote that appears in the unenclosed middle of field(aaa"""bb""cc"), or appears
```

Reply:
Done